### ISSUE TITLE
Cannot render glb files with vsgviewer

### ISSUE THREAD

**Issue by Joaopmoliveira** [vsg-dev/vsgExamples#226]
The file I annexed 
[Link4.zip](https://github.com/vsg-dev/vsgExamples/files/11675825/Link4.zip)
Can be rendered with the Khronos Viewer, but the vsgviewer does not render anything. I have tried to convert it to  the vsgt format with vsgconv tool, which creates a valid file, 
[model.zip](https://github.com/vsg-dev/vsgExamples/files/11675837/model.zip)
But this model is also not rendered. I am experiencing this phenomena with all 3D formats, ply, glft... 
I checked and the vsg::read<Node> call returns a valid pointer. Is this an assimp problem or a vsgXchange problem?

**Comment by vsg-dev** [comment]
I have just tried your Link4.glb and don't see anything on screen when I run:

`vsgviewer Link4.glb
`
Converting to .vsgt with:
`
vsgconv Link4.glb test.vsgt`

And this test.vsgt works without errors, even with vulkan debug enabled:

`vsgviewer test.vsgt -d`

The test.vsgt doesn't have any obvious errors in it so the vsgXchange::assimp does look to be doing something half sensible,. As to the reason for the missing rendering, I don't know what features Link4.glb uses, it could be something as simple as alpha values being zero.

I have just tried loading the reference models in glTF-Sample-Models and the .gltf and .glb files load and rendering fine.  Could you try these models out?

https://github.com/KhronosGroup/glTF-Sample-Models

`vsgviewer 2.0/FlightHelmet/glTF/FlightHelmet.gltf`

Could you provide a screenshot of what you are expecting for Link4.glb.  Where did the model come from?  

Could you provide a link to other formats that aren't working?

Do the vsgExamples/data work?  i.e.

`vsgviewer data/models/lz.vsgt`

**Comment by Joaopmoliveira** [comment]
I can't do more tests for the next 4 days, but I tested .ply, .glb., .glft and out of the three only ply renders anything usefull but even that is not ok (there is some problem with the colors of the vertices). Will get back to this to provide feedback

**Comment by Joaopmoliveira** [comment]
I was expecting to see something like the following: (this is the glb version of the file rendered in the website https://3dviewer.net/#)
<img width="841" alt="Screenshot 2023-06-11 195313" src="https://github.com/vsg-dev/vsgExamples/assets/44142506/0a50aea2-e307-44d3-883f-7635575bc12a">
Now vsgviewer does work with the models from [glTF-Sample-Models](https://github.com/KhronosGroup/glTF-Sample-Models) as shown in the following figure 
<img width="1264" alt="Screenshot 2023-06-11 200235" src="https://github.com/vsg-dev/vsgExamples/assets/44142506/f94b177a-2825-4da7-9476-fc59f8bc86ac">
I have also tested the examples from vsg and they work, so no problem with my system itself.  This leads me to conclude that its a problem with the way assimp of vsgXchange loads the vertices and other parameters into memory. Do you have any suggestion to fix this problem or give up on this particular model?

**Comment by robertosfield** [comment]
I haven't come across GLTF/GLB files that don't work so far so can't speculate on why Assimp/vsgXchange might not handle your specific GLB files.

You'll need to share your problem GLB files.  May also be worth seeing if there is a GLTF validator that you could run on the files to see if there is something odd about them.

It could simply be that the GLB files you have are valid but just use a GLTF feature that Assimp/vsgXchange doesn't support.

**Comment by Joaopmoliveira** [comment]
For posterity sake (other people might face a similar issue), I was able to solve my problem by using meshlab to create a texture (png) associated with the obj file, and then read this new mesh with VSG.

**Comment by vsg-dev** [comment]
It's good to hear you've found a workaround, but I'd like to get to the bottom of the issue.  All the .glb files I have tested work so I need an example of one that doesn't work, details on how it was created so I can look into what the problem was and figure our a fix for it.

**Comment by Joaopmoliveira** [comment]
I

**Comment by robertosfield** [comment]
@Joaopmoliveira did you find the issue?  If not could you provide a source of glb file that doesn't work so I can look into it?

**Comment by Joaopmoliveira** [comment]
Sorry @robertosfield, lately my PhD is consuming most of my time. I will attach a problematic glb file so that you can at least replicate the problem.

**Comment by Joaopmoliveira** [comment]
Actually the Link4.zip already contains a glb file which is problematic

**Comment by robertosfield** [comment]
I have had a chance to look into why the shared Link4.glb doesn't appear - by converting to .vsgt and then inspect I found the per vertex colours all to have an alpha value of 0.0.  Manually changing the alpha value to 1.0 fixed the rendering issue.

It might be that this model is being loaded incorrectly by Assimp and generating 0's where they should be value alpha values, or could be a bug in vsgXchange::assimp loader not mapping the Assimp data across, or it could simply be that the source model has erronous colour values that the other 3rd party viewer ignore.

I have gone through the official glTF-Sample-Models dataset and the models with per vertex colours look to be mapping across the VSG just fine.

I have now used blender to load the Link4.glb and it loads, but when I export it back to glb it has a different number of vertices, and the colors all come through with an alpha value of 1.0 rather than 0.0 and can be viewed with vsgviewer.

I'm going to close this Issue as it looks like the model if broken.
